csharp/extractor/Semmle.Extraction.CSharp/Entities/LocalVariable.cs

```diff
@@ -23,17 +23,23 @@ public sealed override void WriteQuotedId(EscapingTextWriter trapFile)
         public void PopulateManual(Expression parent, bool isVar)
         {
             var trapFile = Context.TrapWriter.Writer;
-            var (kind, type) = Symbol is ILocalSymbol l
-                ? (l.IsRef ? 3 : l.IsConst ? 2 : 1, l.GetAnnotatedType())
-                : (1, parent.Type);
-            trapFile.localvars(this, kind, Symbol.Name, isVar ? 1 : 0, Type.Create(Context, type).TypeRef, parent);
+            var @var = isVar ? 1 : 0;
 
             if (Symbol is ILocalSymbol local)
             {
+                var kind = local.IsRef ? Kinds.VariableKind.Ref : local.IsConst ? Kinds.VariableKind.Const : Kinds.VariableKind.None;
```

Comment:
I agree with you that it is odd that we store the information twice and I also had the same thought as you, when going through the code. The QL library can "easily" be changed to fetch the ref information from the annotated type.
@hvitved : Do you know of any historical reason for this or should we just go ahead and change it? The change will only cause issues if anyone is using the `localvars` relation directly (it will not be changed in this PR as we need to also have up and downgrade scripts for the change).

Reply:
I think this is simply because type annotations were introduced later.

Reply:
Ok, then it probably makes sense that we cleanup the implementation. I will open an issue.